src/lib.rs

```diff
@@ -168,7 +168,7 @@ pub fn firefox_version(binary: &Path) -> Result<AppVersion, Error>  {
     let mut version = AppVersion::new();
     let mut updated = false;
 
-    if let Some(dir) = ini_path(binary) {
+    if let Some(dir) = ini_path(&binary.canonicalize().unwrap()) {
```

Comment:
This unwrap is bad; in the case the path doesn't exist we probably want to return a MetdataError rather than panicing.